pkg/manager/watch_endpoints.go

```diff
@@ -373,7 +373,7 @@ func (sm *Manager) watchEndpoint(ctx context.Context, id string, service *v1.Ser
 				}
 
 				// If there are no local endpoints, and we had one then remove it and stop the leaderElection
-				if lastKnownGoodEndpoint != "" && sm.config.EnableRoutingTable {
+				if lastKnownGoodEndpoint != "" /*&& sm.config.EnableRoutingTable*/ {
```

Comment:
This shouldn't be gated around routing table mode? @p-strusiewiczsurmacki-mobica

Reply:
I agree.
I think the idea was to hide it behind the table mode feature flag back then.
But after the code got refactored this snippet was just missed to adapt to the new call stack.

Reply:
OK cool, I'll clean the code and add a final commit. I'll take an action to refactor the CI/CD testing to create more errors in future to catch this.